src/libfuncs/pedersen.rs

```diff
@@ -78,74 +79,74 @@ pub fn build_pedersen<'ctx>(
 
     // We must extend to i256 because bswap must be an even number of bytes.
 
-    let op = entry.append_operation(arith::constant(
-        context,
-        IntegerAttribute::new(IntegerType::new(context, 64).into(), 1).into(),
-        location,
-    ));
-    let const_1 = op.result(0)?.into();
-
-    let op = entry.append_operation(
-        OperationBuilder::new("llvm.alloca", location)
-            .add_attributes(&[(
-                Identifier::new(context, "alignment"),
-                IntegerAttribute::new(
-                    IntegerType::new(context, 64).into(),
-                    layout_i256.align().try_into()?,
-                )
-                .into(),
-            )])
-            .add_operands(&[const_1])
-            .add_results(&[llvm::r#type::pointer(i256_ty, 0)])
-            .build()?,
-    );
-    let lhs_ptr = op.result(0)?.into();
-
-    let op = entry.append_operation(
-        OperationBuilder::new("llvm.alloca", location)
-            .add_attributes(&[(
-                Identifier::new(context, "alignment"),
-                IntegerAttribute::new(
-                    IntegerType::new(context, 64).into(),
-                    layout_i256.align().try_into()?,
-                )
-                .into(),
-            )])
-            .add_operands(&[const_1])
-            .add_results(&[llvm::r#type::pointer(i256_ty, 0)])
-            .build()?,
-    );
-    let rhs_ptr = op.result(0)?.into();
-
-    let op = entry.append_operation(
-        OperationBuilder::new("llvm.alloca", location)
-            .add_attributes(&[(
-                Identifier::new(context, "alignment"),
-                IntegerAttribute::new(
-                    IntegerType::new(context, 64).into(),
-                    layout_i256.align().try_into()?,
-                )
-                .into(),
-            )])
-            .add_operands(&[const_1])
-            .add_results(&[llvm::r#type::pointer(i256_ty, 0)])
-            .build()?,
-    );
-    let dst_ptr = op.result(0)?.into();
-
-    let op = entry.append_operation(arith::extui(lhs, i256_ty, location));
-    let lhs_i256 = op.result(0)?.into();
-    let op = entry.append_operation(arith::extui(rhs, i256_ty, location));
-    let rhs_i256 = op.result(0)?.into();
+    let const_1 = entry
+        .const_int_from_type(context, location, value, value_ty)?
+        .into();
```

Comment:
Is this code equivalent to the original? The original creates a 64-bit integer, this one creates a felt (252-bit integer).

Reply:
You are right. I just updated this by creating a new `int64_ty` variable